rasterio/_io.pyx

```diff
@@ -1735,11 +1745,19 @@ cdef class InMemoryRaster:
             self._hds = NULL
 
     def read(self):
-        io_auto(self._image, self.band(1), False)
+        if self._image.ndim == 2:
```

Comment:
@snowman2 since `self._image` is not set until a `write` operation, you should first make sure that `self._image` is defined before accessing props or reading from it.  In this case, raising an `IOError` might be the appropriate thing to do.

Reply:
That is a good point. What are your thoughts on initializing with:
```python
        if image is None:
            image = np.zeros((count, height, width))

        self.write(image)
```

Reply:
Okay, I can add that. Would you still like the InMemoryRaster to initialize an array of zeros if the `image` argument is None and `width` and `height` are provided (see most recent commits - I can remove if this is not desired)?

Reply:
Actually, I realized what case you were talking about and have made the appropriate changes & removed the initialization of image to zeros.